### ISSUE TITLE
"Invalid params" when trying to remove breakpoint

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#529]
Had a breakpoint on 60, stepped till I was on 61 then hit F9 again (I was intending to remove the breakpoint on 60, but it added one on 61)... Hit F9 again and saw this.

<img width="1067" alt="screen shot 2018-02-06 at 08 28 17" src="https://user-images.githubusercontent.com/1078012/35848929-c82bcfc6-0b17-11e8-88a4-6e1e2c82b0b2.png">

[observatory.txt](https://github.com/Dart-Code/Dart-Code/files/1698298/observatory.txt)

**Comment by DanTup** [comment]
This error comes back from Observatory. [Here's a log](https://github.com/Dart-Code/Dart-Code/files/1810618/log.txt).

- Adding and removing break point on 61 works fine.
- Adding and removing break point on 62 works fine.
- Adding and removing break point on 62 *while there's a breakpoint on 61* gives this error.

**Comment by DanTup** [comment]
This is caused by the VM sending back the same breakpoint for the second request (since it's basically the same location as the first) and is not handling it well (Code gave us two breakpoints and so we expected two). Then we try to remove the duplicate twice:

```text
==> {"id":"1344","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/779553244","scriptUri":"/M:/Coding/Applications/flutter/packages/flutter_tools/lib/src/vscode/vscode.dart","line":61}}
==> {"id":"1345","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/779553244","scriptUri":"/M:/Coding/Applications/flutter/packages/flutter_tools/lib/src/vscode/vscode.dart","line":62}}
==> {"id":"1346","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/779553244","scriptUri":"package:flutter_tools/src/vscode/vscode.dart","line":61}}
==> {"id":"1347","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/779553244","scriptUri":"package:flutter_tools/src/vscode/vscode.dart","line":62}}

<== {"jsonrpc":"2.0", "result":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/9","breakpointNumber":9,"resolved":false,"location":{"type":"UnresolvedSourceLocation","scriptUri":"\/M:\/Coding\/Applications\/flutter\/packages\/flutter_tools\/lib\/src\/vscode\/vscode.dart","line":61}},"id":"1344"}
<== {"jsonrpc":"2.0", "result":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/10","breakpointNumber":10,"resolved":false,"location":{"type":"UnresolvedSourceLocation","scriptUri":"\/M:\/Coding\/Applications\/flutter\/packages\/flutter_tools\/lib\/src\/vscode\/vscode.dart","line":62}},"id":"1345"}
<== {"jsonrpc":"2.0", "result":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/11","breakpointNumber":11,"resolved":true,"location":{"type":"SourceLocation","script":{"type":"@Script","fixedId":true,"id":"libraries\/@222005062\/scripts\/package%3Aflutter_tools%2Fsrc%2Fvscode%2Fvscode.dart\/1622416cdf2","uri":"package:flutter_tools\/src\/vscode\/vscode.dart","_kind":"library"},"tokenPos":414}},"id":"1346"}
<== {"jsonrpc":"2.0", "result":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/11","breakpointNumber":11,"resolved":true,"location":{"type":"SourceLocation","script":{"type":"@Script","fixedId":true,"id":"libraries\/@222005062\/scripts\/package%3Aflutter_tools%2Fsrc%2Fvscode%2Fvscode.dart\/1622416cdf2","uri":"package:flutter_tools\/src\/vscode\/vscode.dart","_kind":"library"},"tokenPos":414}},"id":"1347"}

==> {"id":"1348","method":"removeBreakpoint","params":{"isolateId":"isolates/779553244","breakpointId":"breakpoints/9"}}
==> {"id":"1349","method":"removeBreakpoint","params":{"isolateId":"isolates/779553244","breakpointId":"breakpoints/10"}}
==> {"id":"1350","method":"removeBreakpoint","params":{"isolateId":"isolates/779553244","breakpointId":"breakpoints/11"}}
==> {"id":"1351","method":"removeBreakpoint","params":{"isolateId":"isolates/779553244","breakpointId":"breakpoints/11"}}
```

**Comment by DanTup** [comment]
This is looking like a bug job... We have an existing issue that we send multiple BP responses in setBreakPointRequest (because we loop over possible uris), but seems like we also need to teach Code that these breakpoints are the same. I tried passing the breakpoint response back through to `setBreakPointsRequest` and then setting the id to `breakpointNumber` (previously we only set `verified`), however that didn't seem to work.

The correct fix is likely that we fix that the first breakpoint appears on the wrong line - if we returned info when the BP on the first line was added, it should've appeared against the next line (or vice versa) and probably that would fix it.

To do this, we need to be able to get sources and convert from tokenPos to line/column. Committed some WIP to come back to.
